code/modules/bitcoin/bitcoin_miner.dm

```diff
@@ -0,0 +1,132 @@
+		/obj/item/gavelhammer,
+		/obj/item/clothing/neck/tie/black
+	)
+	var/upgrade_state = 1
+	var/bitcoin_balance = 0
+	var/min = 1
+	var/max = 2
+	var/got = 0
+	var/perc = 20
+
+/obj/machinery/bitcoin_miner/wrench_act(mob/living/user, obj/item/tool)
+	. = ..()
+	if(default_unfasten_wrench(user, tool, time = 1.5 SECONDS) == SUCCESSFUL_UNFASTEN)
+		return ITEM_INTERACT_SUCCESS
+
+/obj/machinery/materials_market/screwdriver_act(mob/living/user, obj/item/tool)
+	. = ..()
+	if(default_deconstruction_screwdriver(user, base_icon_state, base_icon_state, tool))
+		return ITEM_INTERACT_SUCCESS
+
+/obj/machinery/bitcoin_miner/update_icon_state()
+	if(!powered() && icon_state == "bit_miner")
+		icon_state = base_icon_state
+		return ..()
+	if(powered() && icon_state == "bit_miner_inactive")
+		icon_state = "bit_miner"
+		return ..()
+
+/obj/machinery/bitcoin_miner/crowbar_act(mob/living/user, obj/item/tool)
+	. = ..()
+	if(default_deconstruction_crowbar(tool))
+		return ITEM_INTERACT_SUCCESS
+
+/obj/machinery/bitcoin_miner/attackby(obj/item/O, mob/user, params)
+	if(is_type_in_list(O, money_insert))
+		var/value = O.get_item_credit_value()
+
+		if(!value)
+			say("Cannot insert nil credits.")
+			playsound(src, 'sound/machines/scanner/scanbuzz.ogg', 25, FALSE)
+			return TRUE
+
+		to_chat(user, span_notice("You have inserted [value] credits into the machine. It seems it mines better now."))
+		playsound(src, 'sound/machines/synth/synth_yes.ogg', 50, FALSE)
+		perc = perc + value/100
+		qdel(O)
+
+		return ..()
+
+	else if(is_type_in_list(O, min_insert))
+		to_chat(user, span_notice("You have inserted [O.name] into the machine"))
+		playsound(src, 'sound/machines/compiler/compiler-stage2.ogg', 50, FALSE)
+		qdel(O)
+		min = min + 1
+
+		return ..()
+
+	else if(is_type_in_list(O, max_insert))
+		to_chat(user, span_notice("You have inserted [O.name] into the machine"))
+		playsound(src, 'sound/machines/compiler/compiler-stage2.ogg', 50, FALSE)
+		qdel(O)
+		max = max + 1
+
+	return ..()
+
+/obj/machinery/bitcoin_miner/examine(mob/user)
+	. = ..()
+	. += "The miner reports [bitcoin_balance] bitcoins are available."
+	. += span_notice("The miner can get between [min] and [max] bitcoin.")
+	. += span_notice("ALT-LMB to withdraw bitcoins.")
+	. += span_notice("Current price: [SSbitcoin.get_price()] credits.")
+
+/obj/machinery/bitcoin_miner/proc/mine()
+	got = rand(min, max)
+	bitcoin_balance = bitcoin_balance + got
+	say("Mined [got] bitcoins!")
+	playsound(src, 'sound/machines/sonar-ping.ogg', 25, FALSE)
+	got = 0
+
+/obj/machinery/bitcoin_miner/process(seconds_per_tick)
+	if(SPT_PROB(10, seconds_per_tick) && powered())
+		mine()
+
+/obj/machinery/bitcoin_miner/click_alt(mob/living/user)
+	if(bitcoin_balance > 0)
+		var/obj/item/holochip/holochip = new (user.drop_location(), bitcoin_balance * SSbitcoin.get_price())
+		user.put_in_hands(holochip)
+		bitcoin_balance = 0
+		playsound(src, 'sound/items/equip/sneakers-equip1.ogg', 50, FALSE)

```

Comment:
why the bitcoin miner putting some jordans on

Reply:
is ok to reuse sounds, if it fits - that's cool